core/src/main/java/org/testcontainers/utility/ConfigurationFileImageNameSubstitutor.java

```diff
@@ -0,0 +1,47 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link ImageNameSubstitutor} which takes replacement image names from configuration.
+ * See {@link TestcontainersConfiguration} for the subset of image names which can be substituted using this mechanism.
+ * <p>
+ * WARNING: this class is not intended to be public, but {@link java.util.ServiceLoader}
```

Comment:
outdated comment

btw, since we delegate to it, perhaps we can make it package private?